setup.cfg

```diff
@@ -15,7 +15,7 @@ profile = black
 combine_as_imports = True
 
 [tool:pytest]
-addopts = -rxXs
+addopts = -rxXs -Werror
```

Comment:
use filterwarnings=error

Reply:
This reminds me that we can use `filterwarnings = error:::httpx` to convert only warnings from `httpx` to errors, so that we will not blocked by `uvicorn`.

Reply:
> use filterwarnings=error

Could you expand on that @graingert?

> to exempt uvicorn warnings, so that we will not be blocked by uvicorn.

Oh, that's a great tip, @j178. I'm not sure what I need to do to inline that here, but will look into it.

Reply:
Ah okay, cribbing an example from here https://github.com/pytest-dev/pytest/issues/4180

```ini
[tool:pytest]
addopts =
  -W error
  -v
filterwarnings =
  error
  # ignored by default
  ignore::ImportWarning
  ignore::PendingDeprecationWarning
  # raised by Cython, usually harmless
  ignore:numpy.dtype size changed:RuntimeWarning
  ignore:numpy.ufunc size changed:RuntimeWarning
  # raised by Werkzeug
  ignore:Request.is_xhr is deprecated:DeprecationWarning
  # raised by Moto
  ignore:Flags not at the start of the expression:DeprecationWarning
```

Reply:
The filterwarnings in option is preferred eg https://github.com/pytest-dev/pytest/blob/main/pyproject.toml#L23